test/no-direct-import.js

```diff
@@ -0,0 +1,38 @@
+import test from 'ava';
+import avaRuleTester from 'eslint-ava-rule-tester';
+import rule from '../rules/no-direct-import';
+
+const ruleTester = avaRuleTester(test, {
+	parserOptions: {
+		sourceType: 'module'
+	}
+});
+
+const error = {
+	ruleId: 'no-direct-import',
+	message: 'Do not reference directly'
+};
+
+ruleTester.run('no-direct-import', rule, {
+	valid: [
+		'const {promisify} = require(\'util\');',
+		'import {promisify} from \'util\';',
+		'const file = require(\'unrestricted\')',
+		'import file from \'unrestricted\'',
+		'const util = myFunction(\'util\')'
+	],
+	invalid: [
+		{
+			code: 'const util = require(\'util\');',
+			errors: [error]
+		},
+		{
+			code: 'import util from \'util\';',
+			errors: [error]
+		},
+		{
+			code: 'import * as util from \'util\';',
+			errors: [error]
+		}
+	]
+});
```

Comment:
This needs a lot more tests, both valid and invalid ones. There are many more ways to import. For example:

- `import {promisify as foo} from \'util\';`
- `import {debuglog, promisify} from \'util\';`
- `import util, {promisify} from \'util\';`
- `const foo = myFunction(\'util\')`
- `const {promisify as foo} = require(\'util\');`

Reply:
> * `const {promisify as foo} = require(\'util\');`
It think it should be `const {promisify: foo} = require(\'util\');`.